src/kioskboard.js

```diff
@@ -642,9 +648,10 @@
             if (backspaceKeyElm) {
               backspaceKeyElm.addEventListener('click', function (e) {
                 e.preventDefault();
-
+                
                 // update the selectionStart
-                theInputSelIndex = input.selectionStart || (input.value || '').length;
+                // Math.min prevents out of bound, theInputSelIndex cannot be greater than the length of theInputValArray
+                theInputSelIndex = Math.min((input.selectionStart || (input.value || '').length),theInputValArray.length);
```

Comment:
These changes should be discarded if my suggestion on the adding part is ok for you.